jupyterhub/templates/scheduling/user-scheduler/deployment.yaml

```diff
@@ -0,0 +1,51 @@
+{{- if .Values.scheduling.userScheduler.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: user-scheduler
+  labels:
+    {{- include "jupyterhub.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.scheduling.userScheduler.replicas }}
+  selector:
+    matchLabels:
+      {{- include "jupyterhub.matchLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- /* Changes here will cause the Deployment to restart the pods. */}}
+        {{- include "jupyterhub.matchLabels" . | nindent 8 }}
+        hub.jupyter.org/pod-kind: "core"
+      annotations:
+        # This lets us autorestart when the configmap changes!
+        checksum/config-map: {{ include (print $.Template.BasePath "/scheduling/user-scheduler/configmap.yaml") . | sha256sum }}
+    spec:
+      {{- if .Values.rbac.enabled }}
+      serviceAccountName: user-scheduler
+      {{- end }}
+      containers:
+        - name: user-scheduler
+          image: {{ include "jupyterhub.scheduler.image" . }}
+          command:
+            - /usr/local/bin/kube-scheduler
+            - --scheduler-name=user-scheduler
+            - --policy-configmap=user-scheduler
+            - --policy-configmap-namespace={{ .Release.Namespace }}
+            - --lock-object-name=user-scheduler
+            - --lock-object-namespace={{ .Release.Namespace }}
+            - --leader-elect-resource-lock=configmaps
+            - --v=100
```

Comment:
The scheduler is configured to support leader election, not that it matters much, but it works anyhow. The scheduler is also contained within the jupyterhub namespace.

I should perhaps lower the verbosity =D

Reply:
Note that this way of declaring what configmap to use was deprecated, without an option available in beta-state (k8s 1.11).

We could go for the new fancy way of doing it, involving a new type of configuration resource pointing to the configmap instead, but then we would end up dependent on k8s 1.11+ while we currently are not (even though we use a kube-scheduler binary tagged 1.11.2)

Reply:
Let's stick with the deprecated way for now. We don't want to be too aggressive with our base versions.

Reply:
Can you add a TODO comment in the code to indicate this for when we do eventually  require 1.11?

Reply:
- [ ] Add a TODO comment in code to indicate this for when we do eventually require